### ISSUE TITLE
Bug in comma operator

### ISSUE THREAD

**Issue by adelespinasse** [PiotrDabkowski/Js2Py#170]
The following statement gives the expected output in Node.js:
```
> var a = {}; (a.a=1), (a.b={a: a.a}); console.log(a);
{ a: 1, b: { a: 1 } }
```
But in js2py, it gives different output:
```
>>> js2py.eval_js('var a = {}; (a.a=1), (a.b={a: a.a}); console.log(a);')
{'a': 1, 'b': {'a': None}}
```
It looks as if the first sub-expression of the comma operator, `(a.a=1)`, has not taken effect when the second, `(a.b={a: a.a})`, is evaluated. The problem does not occur if you replace the comma with a semicolon.

I have been unable to reproduce this issue with simpler assignments, like `(a.a=1), (a.b=a.a)`, so I'm not totally sure I understand the bug, but I think it's pretty clear that there is a bug.

I discovered this when using Babel 7 to compile some ES2016+ code into ES5 and translating that with `js2py.translate_file()` (which made tracking down the problem pretty difficult). Babel apparently sometimes uses the comma operator in places where it wasn't used in the input code.

**Comment by PiotrDabkowski** [comment]
Interesting, your code translates roughly to:

```python
PyJs_Object_0_ = Js({})
var.put(u'a', PyJs_Object_0_)
PyJs_Object_1_ = Js({u'a':var.get(u'a').get(u'a')})
PyJsComma(var.get(u'a').put(u'a', Js(1.0)),var.get(u'a').put(u'b', PyJs_Object_1_))
```

PyJs_Object_1_ is being incorrectly constructed before var.get(u'a').put(u'a', Js(1.0) operation. The easiest solution would be to change last 2 lines to:

```python
PyJs_Object_1_ = lambda: Js({u'a':var.get(u'a').get(u'a')})
PyJsComma(var.get(u'a').put(u'a', Js(1.0)),var.get(u'a').put(u'b', PyJs_Object_1_()))
```
Sorry for inconvenience, will fix this tomorrow.

**Comment by adelespinasse** [comment]
Interesting. I would have expected something more like
```Python
PyJsComma(
  var.get(u'a').put(u'a', Js(1.0)),
  var.get(u'a').put(u'b', Js({u'a':var.get(u'a').get(u'a')}))
)
```
but I just started using py2js a few days ago, so what do I know? Thanks for the very fast response!

**Comment by PiotrDabkowski** [comment]
Fixed in efbfcca

Regarding your suggestion, this is not so simple, at head your example translates now to:
```python
PyJsComma(var.get('a').put('a', Js(1.0)),var.get('a').put('b', Js({'a':var.get('a').get('a')})))
```
so exactly as you wrote.

However, there are some edge cases like getters that make translation much harder. For example

`var x = {get y() { return this.val;}, val: 11};x.y` translates to:
```python
def PyJs_Object_0_():
    PyJs_Object_0_ = Js({'val':Js(11.0)})
    @Js
    def PyJs_anonymous_1_(this, arguments, var=var):
        var = Scope({'this':this, 'arguments':arguments}, var)
        var.registers([])
        return var.get(u"this").get('val')
    PyJs_anonymous_1_._set_name('anonymous')
    PyJs_Object_0_.define_own_property('y', {"get":PyJs_anonymous_1_, "configurable":True, "enumerable":True})
    return PyJs_Object_0_
var.put('x', PyJs_Object_0_())
var.get('x').get('y')
```

**Comment by adelespinasse** [comment]
Note: I think this fix has not been released to the pip repository. (It's ok for me; I have a workaround.)
